### DISCUSSION TITLE
Can't switch between model variations

### DISCUSSION THREAD

**Discussion by Kamil-Jasinski** [open-webui/open-webui#2525] [General]
# Bug Report

## Description

**Bug Summary:**
When downloading multiple variants of the same model (e.g., "llama3" and "llama3:instruct"), only the last downloaded model is available in the chat model dropdown. Previously, it was possible to select from multiple variants of the same model.

**Steps to Reproduce:**

1. Go to Settings -> Models -> "Download model from Ollama.com".
2. Enter the tag "llama3" and download the model.
3. Verify that the model "llama3" is available for use in the chat window.
4. Download another model, this time using the tag "llama3:instruct".
5. Check the chat model dropdown again.

**Expected Behavior:**
Both "llama3" and "llama3:instruct" should be available for selection in the chat model dropdown.

**Actual Behavior:**
Only "llama3:instruct" is available in the chat model dropdown, while "llama3" is no longer an option. However, using the "ollama list" command shows that both models are downloaded and available.

## Environment
- Open WebUI Version: v0.1.125
- Ollama (if applicable): 0.1.38
- Operating System: Windows 11 on WSL2 Ubuntu 20.04
- Browser (if applicable): Google Chrome Version 125.0.6422.77

## Reproduction Details
1. Go to Settings -> Models -> "Download model from Ollama.com".
2. Enter the tag "llama3" and download the model.
3. Confirm that "llama3" is available for use.
4. Download the "llama3:instruct" model.
5. Observe that only "llama3:instruct" is now available in the chat model dropdown, and "llama3" is missing.

**Confirmation:**

- [ yes ] I have read and followed all the instructions provided in the README.md.
- [ yes ] I am on the latest version of both Open WebUI and Ollama.
- [ not needed - empty console ] I have included the browser console logs.
- [ - ] I have included the Docker container logs.

## Logs and Screenshots
**Docker Container Logs:**

> 2024-05-23 17:59:35 [GIN] 2024/05/23 - 15:59:35 | 200 |      34.782µs |       127.0.0.1 | GET      "/api/version"
> 2024-05-23 17:59:35 INFO:     172.17.0.1:54008 - "GET /ollama/api/version HTTP/1.1" 200 OK
> 2024-05-23 17:59:35 INFO:     172.17.0.1:54008 - "GET /api/version/updates HTTP/1.1" 200 OK
> 2024-05-23 17:59:37 INFO:     127.0.0.1:48212 - "GET /health HTTP/1.1" 200 OK
> 2024-05-23 17:59:37 INFO:     172.17.0.1:54008 - "GET /ollama/urls HTTP/1.1" 200 OK
> 2024-05-23 17:59:37 [GIN] 2024/05/23 - 15:59:37 | 200 |        25.1µs |       127.0.0.1 | GET      "/api/version"
> 2024-05-23 17:59:37 INFO:     172.17.0.1:54008 - "GET /ollama/api/version HTTP/1.1" 200 OK
> 2024-05-23 17:59:37 INFO:     172.17.0.1:54008 - "GET /litellm/api/model/info HTTP/1.1" 200 OK
> 2024-05-23 17:59:39 INFO:apps.ollama.main:url: http://localhost:11434
> 2024-05-23 17:59:39 INFO:     172.17.0.1:54008 - "POST /ollama/api/pull/0 HTTP/1.1" 200 OK
> 2024-05-23 18:00:07 INFO:     127.0.0.1:55862 - "GET /health HTTP/1.1" 200 OK
> 2024-05-23 18:00:07 [GIN] 2024/05/23 - 16:00:07 | 200 | 27.255636354s |       127.0.0.1 | POST     "/api/pull"
> 2024-05-23 18:00:16 INFO:     172.17.0.1:53688 - "GET /ollama/urls HTTP/1.1" 200 OK
> 2024-05-23 18:00:16 [GIN] 2024/05/23 - 16:00:16 | 200 |        26.7µs |       127.0.0.1 | GET      "/api/version"
> 2024-05-23 18:00:16 INFO:     172.17.0.1:53688 - "GET /ollama/api/version HTTP/1.1" 200 OK
> 2024-05-23 18:00:16 INFO:     172.17.0.1:53688 - "GET /litellm/api/model/info HTTP/1.1" 200 OK
> 2024-05-23 18:00:19 INFO:apps.ollama.main:url: http://localhost:11434
> 2024-05-23 18:00:19 INFO:     172.17.0.1:53688 - "POST /ollama/api/pull/0 HTTP/1.1" 200 OK
> 2024-05-23 18:00:28 [GIN] 2024/05/23 - 16:00:28 | 200 |  9.563791884s |       127.0.0.1 | POST     "/api/pull"
> 2024-05-23 18:00:28 INFO:apps.openai.main:get_all_models()
> 2024-05-23 18:00:28 INFO:apps.ollama.main:get_all_models()
> 2024-05-23 18:00:28 [GIN] 2024/05/23 - 16:00:28 | 200 |     538.901µs |       127.0.0.1 | GET      "/api/tags"
> 2024-05-23 18:00:28 INFO:     172.17.0.1:53688 - "GET /ollama/api/tags HTTP/1.1" 200 OK
> 2024-05-23 18:00:28 INFO:apps.openai.main:get_all_models()
> 2024-05-23 18:00:28 INFO:     172.17.0.1:43444 - "GET /openai/api/models HTTP/1.1" 200 OK
> 2024-05-23 18:00:28 INFO:     172.17.0.1:43448 - "GET /litellm/api/v1/models HTTP/1.1" 200 OK
> 2024-05-23 18:00:37 INFO:     127.0.0.1:34454 - "GET /health HTTP/1.1" 200 OK
> 2024-05-23 18:00:48 INFO:     172.17.0.1:54330 - "GET /ollama/urls HTTP/1.1" 200 OK
> 2024-05-23 18:00:48 [GIN] 2024/05/23 - 16:00:48 | 200 |        23.7µs |       127.0.0.1 | GET      "/api/version"
> 2024-05-23 18:00:48 INFO:     172.17.0.1:54330 - "GET /ollama/api/version HTTP/1.1" 200 OK
> 2024-05-23 18:00:48 INFO:     172.17.0.1:54330 - "GET /litellm/api/model/info HTTP/1.1" 200 OK
> 2024-05-23 18:00:56 INFO:apps.openai.main:get_all_models()
> 2024-05-23 18:00:56 INFO:apps.ollama.main:get_all_models()
> 2024-05-23 18:00:56 INFO:apps.openai.main:get_all_models()
> 2024-05-23 18:00:56 [GIN] 2024/05/23 - 16:00:56 | 200 |     448.302µs |       127.0.0.1 | GET      "/api/tags"
> 2024-05-23 18:00:56 INFO:     172.17.0.1:51966 - "GET /litellm/api/v1/models HTTP/1.1" 200 OK
> 2024-05-23 18:00:56 INFO:     172.17.0.1:51980 - "GET /openai/api/models HTTP/1.1" 200 OK
> 2024-05-23 18:00:56 INFO:     172.17.0.1:51958 - "GET /ollama/api/tags HTTP/1.1" 200 OK
> 2024-05-23 18:01:07 INFO:     127.0.0.1:42482 - "GET /health HTTP/1.1" 200 OK
> 2024-05-23 18:01:12 INFO:     172.17.0.1:44064 - "GET /c/2f6eb9ee-7cb2-4fd7-82e8-bdf64def11ac HTTP/1.1" 304 Not Modified
> 2024-05-23 18:01:12 INFO:     172.17.0.1:44064 - "GET /favicon.png HTTP/1.1" 304 Not Modified
> 2024-05-23 18:01:12 INFO:     172.17.0.1:44064 - "GET /manifest.json HTTP/1.1" 200 OK
> 2024-05-23 18:01:12 INFO:     172.17.0.1:44064 - "GET /api/config HTTP/1.1" 200 OK
> 2024-05-23 18:01:12 INFO:     172.17.0.1:44064 - "GET /api/v1/auths/ HTTP/1.1" 200 OK
> 2024-05-23 18:01:12 INFO:     172.17.0.1:44064 - "GET /api/changelog HTTP/1.1" 200 OK
> 2024-05-23 18:01:12 INFO:apps.openai.main:get_all_models()
> 2024-05-23 18:01:12 INFO:apps.ollama.main:get_all_models()
> 2024-05-23 18:01:12 [GIN] 2024/05/23 - 16:01:12 | 200 |         550µs |       127.0.0.1 | GET      "/api/tags"
> 2024-05-23 18:01:12 INFO:apps.openai.main:get_all_models()
> 2024-05-23 18:01:12 INFO:     172.17.0.1:44074 - "GET /ollama/api/tags HTTP/1.1" 200 OK
> 2024-05-23 18:01:12 INFO:     172.17.0.1:44076 - "GET /openai/api/models HTTP/1.1" 200 OK
> 2024-05-23 18:01:12 INFO:     172.17.0.1:44090 - "GET /litellm/api/v1/models HTTP/1.1" 200 OK
> 2024-05-23 18:01:12 INFO:     172.17.0.1:44090 - "GET /api/v1/modelfiles/ HTTP/1.1" 200 OK
> 2024-05-23 18:01:12 INFO:     172.17.0.1:44090 - "GET /api/v1/prompts/ HTTP/1.1" 200 OK
> 2024-05-23 18:01:12 INFO:     172.17.0.1:44090 - "GET /api/v1/documents/ HTTP/1.1" 200 OK
> 2024-05-23 18:01:12 INFO:     172.17.0.1:44090 - "GET /api/v1/chats/tags/all HTTP/1.1" 200 OK
> 2024-05-23 18:01:12 INFO:apps.openai.main:get_all_models()
> 2024-05-23 18:01:12 INFO:apps.openai.main:get_all_models()
> 2024-05-23 18:01:12 INFO:apps.ollama.main:get_all_models()
> 2024-05-23 18:01:12 INFO:     172.17.0.1:44076 - "GET /openai/api/models HTTP/1.1" 200 OK
> 2024-05-23 18:01:12 [GIN] 2024/05/23 - 16:01:12 | 200 |       448.7µs |       127.0.0.1 | GET      "/api/tags"
> 2024-05-23 18:01:12 INFO:     172.17.0.1:44074 - "GET /litellm/api/v1/models HTTP/1.1" 200 OK
> 2024-05-23 18:01:12 INFO:     172.17.0.1:44064 - "GET /api/v1/chats/ HTTP/1.1" 200 OK
> 2024-05-23 18:01:12 INFO:     172.17.0.1:44096 - "GET /api/v1/chats/2f6eb9ee-7cb2-4fd7-82e8-bdf64def11ac HTTP/1.1" 200 OK
> 2024-05-23 18:01:12 INFO:     172.17.0.1:44090 - "GET /ollama/api/tags HTTP/1.1" 200 OK
> 2024-05-23 18:01:12 INFO:     172.17.0.1:44090 - "GET /api/v1/chats/2f6eb9ee-7cb2-4fd7-82e8-bdf64def11ac/tags HTTP/1.1" 200 OK
> 2024-05-23 18:01:12 [GIN] 2024/05/23 - 16:01:12 | 200 |        25.6µs |       127.0.0.1 | GET      "/api/version"
> 2024-05-23 18:01:12 INFO:     172.17.0.1:44090 - "GET /ollama/api/version HTTP/1.1" 200 OK
> 2024-05-23 18:01:37 INFO:     127.0.0.1:46680 - "GET /health HTTP/1.1" 200 OK
> 2024-05-23 18:02:05 INFO:     172.17.0.1:37340 - "GET /ollama/urls HTTP/1.1" 200 OK
> 2024-05-23 18:02:05 [GIN] 2024/05/23 - 16:02:05 | 200 |          24µs |       127.0.0.1 | GET      "/api/version"
> 2024-05-23 18:02:05 INFO:     172.17.0.1:37340 - "GET /ollama/api/version HTTP/1.1" 200 OK
> 2024-05-23 18:02:05 INFO:     172.17.0.1:37340 - "GET /litellm/api/model/info HTTP/1.1" 200 OK
> 2024-05-23 18:02:07 INFO:     127.0.0.1:45036 - "GET /health HTTP/1.1" 200 OK
> 2024-05-23 18:02:08 INFO:apps.ollama.main:url: http://localhost:11434
> 2024-05-23 18:02:08 INFO:     172.17.0.1:37340 - "POST /ollama/api/pull/0 HTTP/1.1" 200 OK
> 2024-05-23 18:02:18 [GIN] 2024/05/23 - 16:02:18 | 200 |  9.861337536s |       127.0.0.1 | POST     "/api/pull"
> 2024-05-23 18:02:18 INFO:apps.openai.main:get_all_models()
> 2024-05-23 18:02:18 INFO:apps.ollama.main:get_all_models()
> 2024-05-23 18:02:18 [GIN] 2024/05/23 - 16:02:18 | 200 |       479.4µs |       127.0.0.1 | GET      "/api/tags"
> 2024-05-23 18:02:18 INFO:     172.17.0.1:37340 - "GET /ollama/api/tags HTTP/1.1" 200 OK
> 2024-05-23 18:02:18 INFO:apps.openai.main:get_all_models()
> 2024-05-23 18:02:18 INFO:     172.17.0.1:37436 - "GET /openai/api/models HTTP/1.1" 200 OK
> 2024-05-23 18:02:18 INFO:     172.17.0.1:37444 - "GET /litellm/api/v1/models HTTP/1.1" 200 OK
> 2024-05-23 18:02:37 INFO:     127.0.0.1:45960 - "GET /health HTTP/1.1" 200 OK
> 2024-05-23 18:03:07 INFO:     127.0.0.1:44606 - "GET /health HTTP/1.1" 200 OK
> 2024-05-23 18:03:37 INFO:     127.0.0.1:47474 - "GET /health HTTP/1.1" 200 OK
> 2024-05-23 18:04:07 INFO:     127.0.0.1:55924 - "GET /health HTTP/1.1" 200 OK
> 2024-05-23 18:04:37 INFO:     127.0.0.1:42964 - "GET /health HTTP/1.1" 200 OK
> 2024-05-23 18:04:58 INFO:     172.17.0.1:60428 - "GET /manifest.json HTTP/1.1" 200 OK
> 2024-05-23 18:05:07 INFO:     127.0.0.1:56584 - "GET /health HTTP/1.1" 200 OK

**Screenshots (if applicable):**
Before installing llama3:instruct
![image](https://github.com/open-webui/open-webui/assets/78643287/2be126d6-a2fa-4222-a8d3-17a1b5874d8c)

And after
![image](https://github.com/open-webui/open-webui/assets/78643287/a51163ea-507e-43ca-812f-8f163e28abe3)

"ollama list"
![image](https://github.com/open-webui/open-webui/assets/78643287/4d47259b-4e91-458a-91a3-0a7f8d409c06)

## Installation Method
I installed Ollama and then:
https://github.com/open-webui/open-webui?tab=readme-ov-file#installing-open-webui-with-bundled-ollama-support
**With GPU Support** [Nvidia]

## Additional Information
When using the earlier version v0.1.124, no such problem was encountered.

**Comment by justinh-rahb** [comment]
They have the same hash, same model:

llama3 == llama3:latest == llama3:instruct == llama3:8b

**Comment by tjbck** [comment]
This behaviour occurs when the models are the same model and subsequently share the same id. (llama3:instruct and llama3:8b are the same model 😅)

**Reply by Kamil-Jasinski** [reply]
Okay, this is strange to me because when I download, for example, the "**mistral**" **7b and instruct** models, I get two separate models to choose from:
![image](https://github.com/open-webui/open-webui/assets/78643287/fecd6b68-cb56-4141-b548-f79c3aec5987)
![image](https://github.com/open-webui/open-webui/assets/78643287/31612417-f569-4337-8d4d-71a832ca8963)

Why isn't the llama3 instruct model a separate model? From what I understand, instruct models should be different from base/chat models. Where is the issue? Am I misunderstanding the tags, or is there a mistake on ollama.com?
Why do these models have the same SHA? Please explain.

From what I can see, models with different quantizations have different sizes in GB, but here all the versions visible in the screenshot seem to be the same:
![image](https://github.com/open-webui/open-webui/assets/78643287/935cb6f6-2c95-497f-8786-d1680bc129f0)

And yet, despite this, **llama3 8b and instruct** models have a similar situation to **mistral** models I only have one of these models available.
![image](https://github.com/open-webui/open-webui/assets/78643287/2d5a4291-af83-412e-bc34-747753993525)



In addition, testing on the old configuration (with release v0.1.124) it seems that the **llama3:instruct** model performed better in programming compared to the base **llama3:8b** model.
I hope it's not just an illusion that one model is better than the other, but looking for information on the Internet there is a clear distinction between the types of different models.

**Reply by tjbck** [reply]
Strange, It seems like Ollama might've changed their model list behaviour then, I'd suggest you to check if you can reproduce the same behaviour with Ollama CLI.
